### PR TITLE
remove kubenet_template for kubenet with calico

### PR DESCRIPTION
**Pull request by eriksywu**
Currently agentbaker is incorrectly initializing the "vanilla" kubenet CNI setup for a kubenet cluster with calico policies enabled. After the calico-node daemonsets takes over as the CNI provider, its stateful ipam tracking will get out of sync on nodes that have are running non-hostnetwork pods (i.e coredns, corednsautoscaler and metrics-server). This will result in a potential duplicated pod IP.

We should just wait for the calico-node daemonset to provision its CNI chain and not rely on our own vanilla cni when the node starts up.

This also has the added benefit of avoiding the SNAT issue here that's caused by our vanilla CNI config being too trigger happy with MASQ rules (Azure/AKS#2031)

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -454,6 +454,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			}
 			return ""
 		},
+		"HasCalicoNetworkPolicy": func() bool {
+			return cs.Properties.OrchestratorProfile.KubernetesConfig.NetworkPolicy == NetworkPolicyCalico
+		},
 		"HasCiliumNetworkPlugin": func() bool {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin == NetworkPluginCilium
```
Comment (paulgmiller) [inline pkg/agent/baker.go:461]: Looks like cillum has already wormed into agent baker @matt-stam

**Comment by xuto2** [comment]
please add a bit context for future reference and for people without the context

File: pkg/agent/baker_test.go [hunk 2]
```diff
@@ -253,6 +253,14 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 			}
 			config.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = NetworkPluginKubenet
 		}),
+		Entry("AKSUbuntu1804 with containerd and kubenet cni and calico policy", "AKSUbuntu1804+Containerd+Kubenet+Calico", "1.18.2", func(config *datamodel.NodeBootstrappingConfiguration) {
+			config.ContainerService.Properties.AgentPoolProfiles[0].KubernetesConfig = &datamodel.KubernetesConfig{
+				KubeletConfig:    map[string]string{},
+				ContainerRuntime: datamodel.Containerd,
+			}
+			config.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = NetworkPluginKubenet
+			config.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = NetworkPolicyCalico
```
Comment (xuto2) [inline pkg/agent/baker_test.go:262]: s/NetworkPlugin/NetworkPolicy/ ?

**Comment by xuto2** [comment]
a couple more comments otherwise lgtm.

**Comment by eriksywu** [comment]
Confirmed this change addresses the observed networkpolicy issues for kubenet+calico clusters via both automated e2e (internal) and manual testing with namespace-based networkpolicies.

**Pull request merged**